### ISSUE TITLE
Issue with gatsby

### ISSUE THREAD

**Issue by Ananthakrishnanj** [sag1v/react-elastic-carousel#107]
**Describe the bug**
The pages where ever the react elastic carousel is used in my gatsby project the page just reloads again and again causing the page to be inactive

**To Reproduce**
1. Go to any page having Carousel in gatsby application
2. The page will reload infinitily

**Expected behavior**
The page should not reload and should render the caoursel

**Comment by sag1v** [comment]
@Ananthakrishnanj Thanks. I can't reproduce this behavior, [the docs](https://sag1v.github.io/react-elastic-carousel/) are built with gatsby and i don't see the issue you are describing. 

Can you share an example?

**Comment by BasitEllahi** [comment]
HI, i have the same problem. on localhost. its reloading infinitily on the homepage

**Comment by sag1v** [comment]
@BasitEllahi can you create a small git repo so i can run it locally and see the issue?

**Comment by BasitEllahi** [comment]
https://we.tl/t-GnU1ovg7hJ

**Comment by BasitEllahi** [comment]
Here is a preview mp4 of the issue

**Comment by sag1v** [comment]
Thanks, though its hard to understand the issue without a code example or a repo

**Comment by BasitEllahi** [comment]
import React from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import styled from 'styled-components'
import { colors, media, easings, fontsizes } from '../../../lib/style-utils'
import Carousel from 'react-elastic-carousel'

/* General styles for the tab list items */
const TabsContainer = styled.div`
    li{
        font-weight: bold;
        color: ${colors.grey};
        padding-bottom: 15px;
        transition: color ${easings.quick};
    }
    li:hover{
        cursor: pointer;
        color: ${colors.orange};
        transition: color ${easings.quick};
    }
    li.react-tabs__tab--selected{
        color: ${colors.orange};
    }
    div.react-tabs__tab-panel{
        opacity: 0;
        transition: opacity ${easings.slow};
    }
    div.react-tabs__tab-panel.react-tabs__tab-panel--selected{
        opacity: 1;
        transition: opacity ${easings.slow};
    }
    .only_sm{
        display: flex;
    }
    .only_desk{
        display: none;
    }
    button.rec-arrow{
        display: none;
    }
    .carousel_item_inner p{
        margin-bottom: 0;
    }
    .rec-pagination{
        padding: 1rem 0 1.6rem 0;
    }
    .rec-pagination button{
        box-shadow: none;
        background-color: ${colors.orange};
        transition: background-color ${easings.normal};
    }
    .rec-pagination button.rec-dot_active{
        background-color: ${colors.white};
        transition: background-color ${easings.normal};
    }
    ${media.midDesktop`
    .only_sm{
        display: none;
    }
        .only_desk{
            display: block;
        }
    `}
`
const BoxRounded = styled.div`
    width: 100%;
    max-width: 90%;
    margin: 0 auto;
    height: auto;
    background-color: ${colors.white};
    border-radius: 1.5rem;
    padding: 2rem;

    ${media.midDesktop`
        border-radius: 10rem 1.25rem 10rem 1.25rem;
        padding: 9rem 8rem 6rem 8rem;
    `};

    ${media.lgDesktop`
        padding: 10rem 10rem 7rem 7rem;
    `};
`

/* Tab list items wrapper */
const TabListWrapper  = styled.div`
`

/* Tab content wrapper */
const TabContentWrapper = styled.div`
    padding: 0 0;
    h2{
        font-size: ${fontsizes.fs_title_small};
        font-weight: 900;
        margin-bottom: 2rem;;
    }
    ${media.phablet`
        h2{
            font-size: ${fontsizes.fs_title_small};
        }
    `}
    ${media.desktop`
        h2{
            font-size: ${fontsizes.fs_title_medium};
        }
    `}
`

/* Tab Counter */
const TabBox = styled(Tab)`
    :focus {outline:0;}
`

const TabCounter = styled.span`
    color: ${colors.main};
    display: inline-block;
    margin-right: 13px;
    :focus {outline:0;}
`

/* The Paragraph element to split columns */
const Paragraph = styled.p`
    font-size: ${fontsizes.fs_text_extrasmall};
    margin-bottom: 0;
    ${media.lgDesktop`
        column-count: 2;
        column-gap: 45px;
    `};
`

/*Carousel */
const LinkWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: auto;
    padding: 1.6rem 0 1.6rem 0;
    text-align: center;
`
const Link = styled.a`
    display: inline-block;
    width: auto;
    padding: 0.4rem 0.6rem;
    color: ${colors.white};
    font-weight: bold;

`

const WorkTabs = () => {
    /* useState for tab active or not */
        return (
            <TabsContainer className="tab_container">
                <div className="only_desk">
                    <BoxRounded>
                        <Tabs defaultIndex={0} onSelect={index => console.log("Active tab: "+index)}>
                            <div className="grid grid-cols-3">
                                <TabListWrapper className="col-span-3 sm:col-span-3 md:col-span-3 lg:col-span-1 xl:col-span-1">
                                    <TabList>
                                        <TabBox><TabCounter>01.</TabCounter>We Think</TabBox>
                                        <TabBox><TabCounter>02.</TabCounter>We Create</TabBox>
                                        <TabBox><TabCounter>03.</TabCounter>We Connect</TabBox>
                                        <TabBox><TabCounter>04.</TabCounter>We Grow</TabBox>
                                    </TabList>
                                </TabListWrapper>
                                <TabContentWrapper className="col-span-3 sm:col-span-3 md:col-span-3 lg:col-span-2 xl:col-span-2">
                                    <TabPanel>
                                        <h2>Strategies that deliver business results.</h2>
                                        <Paragraph>
                                            Consumers are always connected and approached by brands 24/7. Today, more than ever, it’s important to reach consumers that are most open to your message. We think about how to reach those consumers that are important to your brand and how to make your brand deliver great business results.
                                        </Paragraph>
                                    </TabPanel>
                                    <TabPanel>
                                        <h2>Content that makes your brand matter.</h2>
                                        <Paragraph>
                                            At Superkraft we create content that makes your brand relevant for consumers. We make your brand shine and touch consumers right in their hearts. We create moments that matter for your audience and deliver results for your brand.
                                        </Paragraph>
                                    </TabPanel>
                                    <TabPanel>
                                        <h2>Building meaningful brand connections.</h2>
                                        <Paragraph>
                                            We build brand connections with your audience on Social Media, via CRM, search campaigns ... Relevancy is always the way to go. That’s why our media buyers and planners work closely with creative teams and performance marketeers.
                                        </Paragraph>
                                    </TabPanel>
                                    <TabPanel>
                                        <h2>Growing brands and business results.</h2>
                                        <Paragraph>
                                            At Superkraft we grow brands! Consumers are always connected and brands approach them in that way, making competition harder. With digital strategies, technology, design and performance optimization we grow your business results.
                                        </Paragraph>
                                    </TabPanel>
                                </TabContentWrapper>
                            </div>
                        </Tabs>
                    </BoxRounded>
                </div>
                <div className="only_sm">
                    <div>
                        <LinkWrapper>
                            <Link>
                                We Think
                            </Link>
                            <Link>
                                We Create
                            </Link>
                            <Link>
                                We Connect
                            </Link>
                            <Link>
                                We Grow
                            </Link>
                        </LinkWrapper>
                    </div>
                    <Carousel easing="cubic-bezier(0.65, 0, 0.35, 1)" tiltEasing="cubic-bezier(0.65, 0, 0.35, 1)" transitionMs={800} itemsToShow={1} enableAutoPlay autoPlaySpeed={5000}>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Strategies that deliver business results.</h2>
                            <p>
                                Consumers are always connected and approached by brands 24/7. Today, more than ever, it’s important to reach consumers that are most open to your message. We think about how to reach those consumers that are important to your brand and how to make your brand deliver great business results.
                            </p>
                        </BoxRounded>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Content that makes your brand matter.</h2>
                            <p>
                                At Superkraft we create content that makes your brand relevant for consumers. We make your brand shine and touch consumers right in their hearts. We create moments that matter for your audience and deliver results for your brand.
                            </p>
                        </BoxRounded>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Building meaningful brand connections.</h2>
                            <p>
                                We build brand connections with your audience on Social Media, via CRM, search campaigns ... Relevancy is always the way to go. That’s why our media buyers and planners work closely with creative teams and performance marketeers.
                            </p>
                        </BoxRounded>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Growing brands and business results.</h2>
                            <p>
                                At Superkraft we grow brands! Consumers are always connected and brands approach them in that way, making competition harder. With digital strategies, technology, design and performance optimization we grow your business results.
                            </p>
                        </BoxRounded>
                    </Carousel>
                </div>
            </TabsContainer>
        )
}

export default WorkTabs`

**Comment by BasitEllahi** [comment]
`import React from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import styled from 'styled-components'
import { colors, media, easings, fontsizes } from '../../../lib/style-utils'
import Carousel from 'react-elastic-carousel'

/* General styles for the tab list items */
const TabsContainer = styled.div`
    li{
        font-weight: bold;
        color: ${colors.grey};
        padding-bottom: 15px;
        transition: color ${easings.quick};
    }
    li:hover{
        cursor: pointer;
        color: ${colors.orange};
        transition: color ${easings.quick};
    }
    li.react-tabs__tab--selected{
        color: ${colors.orange};
    }
    div.react-tabs__tab-panel{
        opacity: 0;
        transition: opacity ${easings.slow};
    }
    div.react-tabs__tab-panel.react-tabs__tab-panel--selected{
        opacity: 1;
        transition: opacity ${easings.slow};
    }
    .only_sm{
        display: flex;
    }
    .only_desk{
        display: none;
    }
    button.rec-arrow{
        display: none;
    }
    .carousel_item_inner p{
        margin-bottom: 0;
    }
    .rec-pagination{
        padding: 1rem 0 1.6rem 0;
    }
    .rec-pagination button{
        box-shadow: none;
        background-color: ${colors.orange};
        transition: background-color ${easings.normal};
    }
    .rec-pagination button.rec-dot_active{
        background-color: ${colors.white};
        transition: background-color ${easings.normal};
    }
    ${media.midDesktop`
    .only_sm{
        display: none;
    }
        .only_desk{
            display: block;
        }
    `}
`
const BoxRounded = styled.div`
    width: 100%;
    max-width: 90%;
    margin: 0 auto;
    height: auto;
    background-color: ${colors.white};
    border-radius: 1.5rem;
    padding: 2rem;

    ${media.midDesktop`
        border-radius: 10rem 1.25rem 10rem 1.25rem;
        padding: 9rem 8rem 6rem 8rem;
    `};

    ${media.lgDesktop`
        padding: 10rem 10rem 7rem 7rem;
    `};
`

/* Tab list items wrapper */
const TabListWrapper  = styled.div`
`

/* Tab content wrapper */
const TabContentWrapper = styled.div`
    padding: 0 0;
    h2{
        font-size: ${fontsizes.fs_title_small};
        font-weight: 900;
        margin-bottom: 2rem;;
    }
    ${media.phablet`
        h2{
            font-size: ${fontsizes.fs_title_small};
        }
    `}
    ${media.desktop`
        h2{
            font-size: ${fontsizes.fs_title_medium};
        }
    `}
`

/* Tab Counter */
const TabBox = styled(Tab)`
    :focus {outline:0;}
`

const TabCounter = styled.span`
    color: ${colors.main};
    display: inline-block;
    margin-right: 13px;
    :focus {outline:0;}
`

/* The Paragraph element to split columns */
const Paragraph = styled.p`
    font-size: ${fontsizes.fs_text_extrasmall};
    margin-bottom: 0;
    ${media.lgDesktop`
        column-count: 2;
        column-gap: 45px;
    `};
`

/*Carousel */
const LinkWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: auto;
    padding: 1.6rem 0 1.6rem 0;
    text-align: center;
`
const Link = styled.a`
    display: inline-block;
    width: auto;
    padding: 0.4rem 0.6rem;
    color: ${colors.white};
    font-weight: bold;

`

const WorkTabs = () => {
    /* useState for tab active or not */
        return (
            <TabsContainer className="tab_container">
                <div className="only_desk">
                    <BoxRounded>
                        <Tabs defaultIndex={0} onSelect={index => console.log("Active tab: "+index)}>
                            <div className="grid grid-cols-3">
                                <TabListWrapper className="col-span-3 sm:col-span-3 md:col-span-3 lg:col-span-1 xl:col-span-1">
                                    <TabList>
                                        <TabBox><TabCounter>01.</TabCounter>We Think</TabBox>
                                        <TabBox><TabCounter>02.</TabCounter>We Create</TabBox>
                                        <TabBox><TabCounter>03.</TabCounter>We Connect</TabBox>
                                        <TabBox><TabCounter>04.</TabCounter>We Grow</TabBox>
                                    </TabList>
                                </TabListWrapper>
                                <TabContentWrapper className="col-span-3 sm:col-span-3 md:col-span-3 lg:col-span-2 xl:col-span-2">
                                    <TabPanel>
                                        <h2>Strategies that deliver business results.</h2>
                                        <Paragraph>
                                            Consumers are always connected and approached by brands 24/7. Today, more than ever, it’s important to reach consumers that are most open to your message. We think about how to reach those consumers that are important to your brand and how to make your brand deliver great business results.
                                        </Paragraph>
                                    </TabPanel>
                                    <TabPanel>
                                        <h2>Content that makes your brand matter.</h2>
                                        <Paragraph>
                                            At Superkraft we create content that makes your brand relevant for consumers. We make your brand shine and touch consumers right in their hearts. We create moments that matter for your audience and deliver results for your brand.
                                        </Paragraph>
                                    </TabPanel>
                                    <TabPanel>
                                        <h2>Building meaningful brand connections.</h2>
                                        <Paragraph>
                                            We build brand connections with your audience on Social Media, via CRM, search campaigns ... Relevancy is always the way to go. That’s why our media buyers and planners work closely with creative teams and performance marketeers.
                                        </Paragraph>
                                    </TabPanel>
                                    <TabPanel>
                                        <h2>Growing brands and business results.</h2>
                                        <Paragraph>
                                            At Superkraft we grow brands! Consumers are always connected and brands approach them in that way, making competition harder. With digital strategies, technology, design and performance optimization we grow your business results.
                                        </Paragraph>
                                    </TabPanel>
                                </TabContentWrapper>
                            </div>
                        </Tabs>
                    </BoxRounded>
                </div>
                <div className="only_sm">
                    <div>
                        <LinkWrapper>
                            <Link>
                                We Think
                            </Link>
                            <Link>
                                We Create
                            </Link>
                            <Link>
                                We Connect
                            </Link>
                            <Link>
                                We Grow
                            </Link>
                        </LinkWrapper>
                    </div>
                    <Carousel easing="cubic-bezier(0.65, 0, 0.35, 1)" tiltEasing="cubic-bezier(0.65, 0, 0.35, 1)" transitionMs={800} itemsToShow={1} enableAutoPlay autoPlaySpeed={5000}>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Strategies that deliver business results.</h2>
                            <p>
                                Consumers are always connected and approached by brands 24/7. Today, more than ever, it’s important to reach consumers that are most open to your message. We think about how to reach those consumers that are important to your brand and how to make your brand deliver great business results.
                            </p>
                        </BoxRounded>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Content that makes your brand matter.</h2>
                            <p>
                                At Superkraft we create content that makes your brand relevant for consumers. We make your brand shine and touch consumers right in their hearts. We create moments that matter for your audience and deliver results for your brand.
                            </p>
                        </BoxRounded>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Building meaningful brand connections.</h2>
                            <p>
                                We build brand connections with your audience on Social Media, via CRM, search campaigns ... Relevancy is always the way to go. That’s why our media buyers and planners work closely with creative teams and performance marketeers.
                            </p>
                        </BoxRounded>
                        <BoxRounded className="carousel_item_inner">
                            <h2>Growing brands and business results.</h2>
                            <p>
                                At Superkraft we grow brands! Consumers are always connected and brands approach them in that way, making competition harder. With digital strategies, technology, design and performance optimization we grow your business results.
                            </p>
                        </BoxRounded>
                    </Carousel>
                </div>
            </TabsContainer>
        )
}

export default WorkTabs`

**Comment by BasitEllahi** [comment]
[carrouseel.js.zip](https://github.com/sag1v/react-elastic-carousel/files/5530172/carrouseel.js.zip)

**Comment by BasitEllahi** [comment]
Here's a the component where i implemented the react elastic carrousel

**Comment by BasitEllahi** [comment]
in package.json using this version  "react-elastic-carousel": "^0.7.5"

**Comment by sag1v** [comment]
Can you check it with latest version (0.9.4)?

**Comment by BasitEllahi** [comment]
yes i updated now, but its stil has the same issue

**Comment by sag1v** [comment]
Its probably related to a SSR issue with the component. I will try to reproduce it locally and tackle it.
Did you use a gatsby starter repo?

**Comment by sag1v** [comment]
Hi @Ananthakrishnanj & @BasitEllahi I think i solved it. 
Can you install version `0.9.5-beta.1` and check it it works as expected?

Just do 
```bash
npm i react-elastic-carousel@0.9.5-beta.1
```
 or
```bash
npm i react-elastic-carousel@next
```

I want to be sure it works for you before i merge to master.

**Comment by BasitEllahi** [comment]
Hi, thanks, its working now. i dont have any issue's with the infinty reloading. i did it with 
yarn add react-elastic-carousel@next

**Comment by sag1v** [comment]
Great! Lets wait for @Ananthakrishnanj to confirm everything is good from his/her side and i will merge this to master and publish for latest version (so you won't need to use beta version)

**Comment by sag1v** [comment]
Fixed as of version `0.9.5`.

Thanks for the catch!
